rskj-core/src/main/java/co/rsk/rpc/netty/JsonRpcWeb3FilterHandler.java

```diff
@@ -67,15 +72,39 @@ protected void channelRead0(ChannelHandlerContext ctx, FullHttpRequest request)
         // the request will be inmediately rejected since it won't be parsed as a valid URI
         // and won't work to match an item on rpc.host
         String hostHeader = headers.get(HttpHeaders.Names.HOST);
-        String parsedHeader = parseHostHeader(hostHeader);
 
-        if (!acceptedHosts.contains(parsedHeader)) {
-            logger.debug("Invalid header HOST {}", hostHeader);
-            response = new DefaultHttpResponse(HttpVersion.HTTP_1_1, HttpResponseStatus.BAD_REQUEST);
-            ctx.write(response).addListener(ChannelFutureListener.CLOSE);
+        if (hostHeader == null || hostHeader.isEmpty()) {
+            this.serveRequest(ctx, request);
+            return;
+        }
+
+        String host = parseHostHeader(hostHeader);
```

Comment:
This could me moved closer to where it is used, it is not used in next `if` statement